### PR TITLE
Clone reverse fixes

### PR DESCRIPTION
**Pull request by ricardoquesada**
2nd part of big patch.
- `copyWithZone()` replaced by `clone()`
  `clone()` already returns an autoreleased object. So, no need to call `autorelease` manually.
- `reverse()` is const, and in some cases is pure virtual function.

examples updated.
Added missing `clone()` and `reverse()`

### CODE REVIEW

**Comment by dumganhar** [comment]
Merged at https://github.com/cocos2d/cocos2d-x/pull/2922

**Comment by dumganhar** [comment]
Thanks.

File: cocos2dx/actions/CCActionGrid.cpp [hunk 1]
```diff
@@ -93,6 +75,11 @@ void CCGridAction::startWithTarget(CCNode *pTarget)
     }
 }
 
+CCGridAction* CCGridAction::reverse() const
+{
+	return (CCGridAction*)CCReverseTime::create( this->clone() );
```
Comment (dumganhar) [inline cocos2dx/actions/CCActionGrid.cpp:80]: Is `CCReverseTime` an instance of `CCGridAction` ?

Comment (ricardoquesada) [inline cocos2dx/actions/CCActionGrid.cpp:80]: Nope. Actually that could trigger a potential bug.
I should add the XXX there.

**Pull request merged**